### ISSUE TITLE
Misunderstanding fit_generator()

### ISSUE THREAD

**Issue by Lucky38i** [keras-team/keras#13560]
**System information**  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  Yes
- TensorFlow-gpu version: 2.0.0  
- Keras version:  2.3.1
- Python version:  3.7
- CUDA/cuDNN version: 10  
- GPU model and memory:  GTX 1070 w/ 8GB

**Describe the current behavior**  
So I'm not sure how fit_generator() interacts with the ImageDataGenerator. Whether in the Epoch logs it's 1,890 batches per epoch or 1,890 images being looked at. I've attached code that I've used below

**Describe the expected behavior**  
Upon training the model after 30 epochs it produces a 99% accuracy to the validate set however when predicting unseen images it fails almost every time.

**Code to reproduce the issue**  
Loading dataset: 
   ```
 batch_size = 32
    epochs = 200

    train_datagen = ImageDataGenerator(rotation_range=40,
                                       width_shift_range=0.2,
                                       height_shift_range=0.2,
                                       rescale=1 / 255,
                                       shear_range=0.2,
                                       zoom_range=0.2,
                                       horizontal_flip=True,
                                       fill_mode='nearest',
                                       validation_split=0.2)

    test_datagen = ImageDataGenerator(rescale=1 / 255)

    train_generator = train_datagen.flow_from_directory(train_dir, target_size=(100, 100),
                                                        batch_size=batch_size, shuffle=True)
    test_generator = test_datagen.flow_from_directory(test_dir, target_size=(100, 100),
                                                      batch_size=batch_size, `shuffle=True)`
```
Training Model:
   ```
 callbacks = [ModelCheckpoint(filepath=checkpoint_file, verbose=1, save_best_only=True, save_weights_only=False)]

    model.fit_generator(train_generator,
                        steps_per_epoch=train_generator.samples // batch_size,
                        epochs=epochs,
                        verbose=1,
                        callbacks=callbacks,
                        validation_data=test_generator,
                        validation_steps=test_generator.samples // batch_size,
                        shuffle=True)
```
Loading the model and testing on unseen image (In a different instance):
```
def preprocess_image(file_dir, show=False):
    temp_img = image.load_img(file_dir, target_size=(100, 100))
    temp_img = img_to_array(temp_img)
    temp_img = temp_img.reshape((1,) + temp_img.shape)
    temp_img = temp_img/255.

    if show:
        plt.imshow(temp_img[0])
        plt.axis('off')
        plt.show()

    return temp_img
test_datagen = ImageDataGenerator(rescale=1 / 255)
test_generator = test_datagen.flow_from_directory(test_dir, target_size=(100, 100)) label_map = test_generator.class_indices
model = load_model(checkpoint_file)

img_tensor = preprocess_image("data/upload/apples1.jpg")

classes = model.predict_classes(img_tensor, batch_size=1)
for label, num in label_map.items():
    if num == classes:
        print("I think this is a:", label)
```

**Other info / logs**  
I'm using this dataset: https://www.kaggle.com/moltean/fruits

Epoch Logs:
```
 897/1890 [=============>................] - ETA: 1:04 - loss: 0.0734 - accuracy: 0.9831
 898/1890 [=============>................] - ETA: 1:04 - loss: 0.0733 - accuracy: 0.9831
 899/1890 [=============>................] - ETA: 1:04 - loss: 0.0732 - accuracy: 0.9831
 900/1890 [=============>................] - ETA: 1:04 - loss: 0.0731 - accuracy: 0.9831
 901/1890 [=============>................] - ETA: 1:04 - loss: 0.0731 - accuracy: 0.9831
 902/1890 [=============>................] - ETA: 1:04 - loss: 0.0730 - accuracy: 0.9831
 903/1890 [=============>................] - ETA: 1:04 - loss: 0.0730 - accuracy: 0.9831
```

**Comment by 22preich** [comment]
I don't know whether I'm understanding your question correctly.

If you are simply asking about the console output of model.fit_generator() when in verbose=1 mode, it's is the number of images per epoch. Those are not epochs, they are steps. If you just want to see epoch output, use verbose=2 in model.fir_generator(). Your validation data will be evaluated at the end of the epoch. The accuracy you see in those log outputs are **not** validation accuracies.

> verbose: Integer. 0, 1, or 2. Verbosity mode. 0 = silent, 1 = progress bar, 2 = one line per epoch.
[https://keras.io/models/model/#fit_generator](url)

sample output of an epoch using verbose=2:
`
Epoch 1/15`
`310/310 - 122s - loss: 0.6432 - accuracy: 0.6444 - val_loss: 0.6892 - val_accuracy: 0.5814
`
ignore those values
